### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by YurySemchenko**
First version of role for installation minio

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,64 @@
...
+  - `minio_http_port` - Port for listening connections   
+     default: `9000`
+  - `minio_access_key` - Access key for log in   
+  - `minio_secret_key` - Secret key for log in   
+
+
+Example Playbook
+----------------
+
+---
```
Comment (pavelpikta) [inline README.md]:
```yaml
---
- name: Install Minio storage server
  hosts: minio
  roles:
    - role: ansible-role-minio
  vars:
    minio_access_key: EH3FJ946FA17VFAPAQVB
    minio_secret_key: WiiS0bcJf+jxqgPJkDJT2tGvFbpyCS9E+FipKwO7

```

File: handlers/main.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  service:
+    name: minio
+    state: restarted
+    enabled: True
+  notify: Wait for Minio
+  become: True
+
+- name: Wait for Minio
+  uri:
+    url: "http://localhost:{{ minio_http_port }}/minio/login"
```
Comment (pavelpikta) [inline handlers/main.yml]: Why localhost ?

**Review by pavelpikta — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (YurySemchenko) [reply README.md]: Fixed

File: README.md [hunk 3]
```diff
@@ -1,2 +1,64 @@
-# ansible-role-minio
-Ansible role to install minio
+minio role
+=========
+[![License](https://img.shields.io/badge/license-Apache-green.svg?style=flat)](https://raw.githubusercontent.com/lean-delivery/ansible-role-minio/master/LICENSE)
+[![Build Status](https://travis-ci.org/lean-delivery/ansible-role-minio.svg?branch=master)](https://travis-ci.org/lean-delivery/ansible-role-minio)
+[![Build Status](https://gitlab.com/lean-delivery/ansible-role-minio/badges/master/build.svg)](https://gitlab.com/lean-delivery/ansible-role-minio)
```
Comment (tgadiev) [inline README.md]:
```suggestion
[![Build Status](https://gitlab.com/lean-delivery/ansible-role-minio/badges/master/build.svg)](https://gitlab.com/lean-delivery/ansible-role-minio/pipelines)
```

File: handlers/main.yml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    enabled: True
+  notify: Wait for Minio
+  become: True
+
+- name: Wait for Minio
+  uri:
+    url: "http://{{ minio_hostname }}:{{ minio_http_port }}/minio/login"
+    status_code: 403
+  register: login_page
+  until: login_page.status == 403
```
Comment (tgadiev) [inline handlers/main.yml]: Why 403?

File: handlers/main.yml (hunk 2 above)
Comment (YurySemchenko) [reply handlers/main.yml]: Changed to {{ ansible_fqdn }}

**Review by pavelpikta — APPROVED** [review]

File: meta/main.yml [hunk 5]
```diff
@@ -0,0 +1,12 @@
+---
+galaxy_info:
+  role_name: minio
+  author: Lean Delivery Team <team@lean-delivery.com>
+  description: Yet Another Ansible Role From Lean Delivery
+  company: EPAM Systems
+  issue_tracker_url: https://github.com/lean-delivery/ansible-role-minio/issues
+  license: Apache
+  min_ansible_version: 2.5
+  platforms: [{name: [EL], versions: [7]}]
```
Comment (tgadiev) [inline meta/main.yml]: Ubuntu?

File: tasks/main.yml [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+  - name: Copy minio binary file
+    get_url:
+      url: "{{ minio_package_source }}"
+      dest: "{{ minio_home }}/bin/minio"
+      mode: +x
+    become: True
+
+  - name: Copy minio config
+    template:
+      src: "minio.conf.j2"
```
Comment (tgadiev) [inline tasks/main.yml]:
```suggestion
      src: minio.conf.j2
```

File: tasks/main.yml [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+      path: "{{ minio_home }}"
+      state: directory
+      mode: 0750
+      owner: '{{ minio_user }}'
+      group: "{{ minio_group }}"
+    become: True
+
+  - name: Ensure minio bin directory is created in {{ minio_home }}
+    file:
+      path: "{{ minio_home }}/bin"
```
Comment (tgadiev) [inline tasks/main.yml]:
```suggestion
      path: '{{ minio_home }}/bin'
```

File: tasks/main.yml [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+      shell: /sbin/nologin
+    become: True
+
+  - name: Ensure minio home directory is created
+    file:
+      path: "{{ minio_home }}"
+      state: directory
+      mode: 0750
+      owner: '{{ minio_user }}'
+      group: "{{ minio_group }}"
```
Comment (tgadiev) [inline tasks/main.yml]:
```suggestion
      group: '{{ minio_group }}'
```

File: tasks/main.yml [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+  - name: Create minio user
+    user:
+      name: '{{ minio_user }}'
+      home: '{{ minio_home }}'
+      shell: /sbin/nologin
+    become: True
+
+  - name: Ensure minio home directory is created
+    file:
+      path: "{{ minio_home }}"
```
Comment (tgadiev) [inline tasks/main.yml]:
```suggestion
      path: '{{ minio_home }}'
```

File: tasks/main.yml [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+      src: "minio.conf.j2"
+      dest: '{{ minio_home }}/minio.conf'
+      backup: True
+      mode: 0644
+      owner: '{{ minio_user }}'
+    become: True
+
+  - name: Copy minio service
+    template:
+      src: "minio.service.j2"
```
Comment (tgadiev) [inline tasks/main.yml]:
```suggestion
      src: minio.service.j2
```

File: tasks/main.yml [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+      dest: /etc/systemd/system/minio.service
+      backup: True
+      mode: 0644
+    become: True
+    notify:
+      - Start Minio
+  - meta: flush_handlers
```
Comment (tgadiev) [inline tasks/main.yml]: Why handlers?

File: tasks/main.yml [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+
+  - name: Copy minio config.json
+    template:
+      src: "config.json.j2"
```
Comment (tgadiev) [inline tasks/main.yml]:
```suggestion
      src: config.json.j2
```

File: tasks/main.yml [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+      dest: '{{ minio_home }}/data/.minio.sys/config/config.json'
+      backup: True
+      mode: 0644
+      owner: '{{ minio_user }}'
+    become: True
+    notify:
+      - Restart Minio
+  - meta: flush_handlers
```
Comment (tgadiev) [inline tasks/main.yml]: Why flush handlers?

File: templates/config.json.j2 [hunk 14]
```diff
@@ -0,0 +1,200 @@
+{
+	"version": "33",
+	"credential": {
+		"accessKey": "{{ minio_access_key }}",
+		"secretKey": "{{ minio_secret_key }}",
+		"expiration": "1970-01-01T00:00:00Z",
```
Comment (tgadiev) [inline templates/config.json.j2:6]: ???

**Review by tgadiev — CHANGES_REQUESTED** [review]

File: tasks/main.yml [hunk 15]
```diff
@@ -0,0 +1,77 @@
...
+- name: Create minio user
+  user:
+    name: '{{ minio_user }}'
+    home: '{{ minio_home }}'
+    shell: /sbin/nologin
+  become: True
+
+- name: Ensure minio home directory is created
+  file:
+    path: "{{ minio_home }}"
```
Comment (tgadiev) [inline tasks/main.yml]:
```suggestion
    path: '{{ minio_home }}'
```

File: tasks/main.yml [hunk 16]
```diff
@@ -0,0 +1,77 @@
...
+    state: directory
+    mode: 0750
+    owner: '{{ minio_user }}'
+    group: "{{ minio_group }}"
```
Comment (tgadiev) [inline tasks/main.yml]:
```suggestion
    group: '{{ minio_group }}'
```

File: tasks/main.yml [hunk 17]
```diff
@@ -0,0 +1,77 @@
...
+  become: True
+
+- name: Ensure minio bin directory is created in {{ minio_home }}
+  file:
+    path: "{{ minio_home }}/bin"
```
Comment (tgadiev) [inline tasks/main.yml]:
```suggestion
    path: '{{ minio_home }}/bin'
```

File: tasks/main.yml [hunk 18]
```diff
@@ -0,0 +1,77 @@
...
+    state: directory
+    mode: 0750
+    owner: '{{ minio_user }}'
+    group: "{{ minio_group }}"
```
Comment (tgadiev) [inline tasks/main.yml]:
```suggestion
    group: '{{ minio_group }}'
```

File: tasks/main.yml [hunk 19]
```diff
@@ -0,0 +1,77 @@
...
+  become: True
+
+- name: Ensure minio data directory is created in {{ minio_home }}
+  file:
+    path: "{{ minio_home }}/data"
```
Comment (tgadiev) [inline tasks/main.yml]:
```suggestion
    path: '{{ minio_home }}/data'
```

File: tasks/main.yml [hunk 20]
```diff
@@ -0,0 +1,77 @@
...
+    state: directory
+    mode: 0750
+    owner: '{{ minio_user }}'
+    group: "{{ minio_group }}"
```
Comment (tgadiev) [inline tasks/main.yml]:
```suggestion
    group: '{{ minio_group }}'
```

File: tasks/main.yml [hunk 21]
```diff
@@ -0,0 +1,77 @@
...
+  become: True
+
+- name: Copy minio binary file
+  get_url:
+    url: "{{ minio_package_source }}"
```
Comment (tgadiev) [inline tasks/main.yml]:
```suggestion
    url: '{{ minio_package_source }}'
```

File: tasks/main.yml [hunk 22]
```diff
@@ -0,0 +1,77 @@
...
+    dest: "{{ minio_home }}/bin/minio"
```
Comment (tgadiev) [inline tasks/main.yml]:
```suggestion
    dest: '{{ minio_home }}/bin/minio'
```

File: tasks/main.yml [hunk 23]
```diff
@@ -0,0 +1,77 @@
...
+    mode: +x
+  become: True
+
+- name: Copy minio config
+  template:
+    src: "minio.conf.j2"
```
Comment (tgadiev) [inline tasks/main.yml]:
```suggestion
    src: minio.conf.j2
```

File: tasks/main.yml [hunk 24]
```diff
@@ -0,0 +1,77 @@
...
+    dest: '{{ minio_home }}/minio.conf'
+    backup: True
+    mode: 0644
+    owner: '{{ minio_user }}'
+  become: True
+
+- name: Copy minio service
+  template:
+    src: "minio.service.j2"
```
Comment (tgadiev) [inline tasks/main.yml]:
```suggestion
    src: minio.service.j2
```

File: tasks/main.yml [hunk 25]
```diff
@@ -0,0 +1,77 @@
...
+    dest: /etc/systemd/system/minio.service
+    backup: True
+    mode: 0644
+  become: True
+  notify:
+    - Start Minio
+
+- meta: flush_handlers
```
Comment (tgadiev) [inline tasks/main.yml]: ???

File: tasks/main.yml [hunk 26]
```diff
@@ -0,0 +1,77 @@
...
+
+- name: Copy minio config.json
+  template:
+    src: "config.json.j2"
```
Comment (tgadiev) [inline tasks/main.yml]:
```suggestion
    src: config.json.j2
```

File: tasks/main.yml [hunk 27]
```diff
@@ -0,0 +1,77 @@
...
+    dest: '{{ minio_home }}/data/.minio.sys/config/config.json'
+    backup: True
+    mode: 0644
+    owner: '{{ minio_user }}'
+  become: True
+  notify:
+    - Restart Minio
+
+- meta: flush_handlers
```
Comment (tgadiev) [inline tasks/main.yml]: ???

**Review by tgadiev — CHANGES_REQUESTED** [review]

File: templates/config.json.j2 (hunk 14 above)
Comment (YurySemchenko) [reply templates/config.json.j2:6]: This is the credentials for accessing Minio. And in this case we install it ourselves.
Expiration is left in the default state. I understand that if this date does not come, then the expiration date will also not come ))

File: tasks/main.yml (hunk 23 above)
Comment (YurySemchenko) [reply tasks/main.yml]: Fixed

File: tasks/main.yml (hunk 24 above)
Comment (YurySemchenko) [reply tasks/main.yml]: Fixed

File: defaults/main.yml [hunk 28]
```diff
@@ -0,0 +1,11 @@
+---
+# defaults file for aem-minio
+minio_home: /opt/minio
+minio_user: minio
+minio_group: minio
+minio_package_source: https://dl.minio.io/server/minio/release/linux-amd64/minio
+data_path: '{{ minio_home }}/data'
```
Comment (tgadiev) [inline defaults/main.yml]:
```suggestion
minio_data_path: '{{ minio_home }}/data'
```

File: meta/main.yml [hunk 29]
```diff
@@ -0,0 +1,25 @@
+---
+galaxy_info:
+  role_name: minio
+  author: Lean Delivery Team <team@lean-delivery.com>
+  description: Lean Delivery MinIO object storage install
+  company: EPAM Systems
+  issue_tracker_url: https://github.com/lean-delivery/ansible-role-minio/issues
+  license: Apache
+  min_ansible_version: 2.5
+  platforms:
```
Comment (tgadiev) [inline meta/main.yml:10]: Debian?

File: tasks/main.yml [hunk 30]
```diff
@@ -0,0 +1,77 @@
...
+    path: '{{ minio_home }}/bin'
+    state: directory
+    mode: 0750
+    owner: '{{ minio_user }}'
+    group: '{{ minio_group }}'
+  become: True
+
+- name: Ensure minio data directory is created in {{ minio_home }}
+  file:
+    path: '{{ minio_home }}/data'
```
Comment (tgadiev) [inline tasks/main.yml]:
```suggestion
    path: '{{ minio_data_path }}'
```

File: tasks/main.yml [hunk 31]
```diff
@@ -0,0 +1,77 @@
...
+- name: Copy minio binary file
+  get_url:
+    url: '{{ minio_package_source }}'
+    dest: '{{ minio_home }}/bin/minio'
+    mode: +x
+  become: True
+
+- name: Copy minio config
+  template:
+    src: 'minio.conf.j2'
```
Comment (tgadiev) [inline tasks/main.yml]:
```suggestion
    src: minio.conf.j2
```

**Review by tgadiev — CHANGES_REQUESTED** [review]

File: handlers/main.yml (hunk 4 above)
Comment (YurySemchenko) [reply handlers/main.yml]: Now this check removed

File: tasks/main.yml (hunk 20 above)
Comment (YurySemchenko) [reply tasks/main.yml]: Fixed

File: tasks/main.yml (hunk 21 above)
Comment (YurySemchenko) [reply tasks/main.yml]: Fixed

File: tasks/main.yml (hunk 22 above)
Comment (YurySemchenko) [reply tasks/main.yml]: Fixed

File: tasks/main.yml (hunk 25 above)
Comment (YurySemchenko) [reply tasks/main.yml]: Now this mechanism is changed

File: tasks/main.yml (hunk 27 above)
Comment (YurySemchenko) [reply tasks/main.yml]: Now this mechanism is changed

File: meta/main.yml (hunk 29 above)
Comment (YurySemchenko) [reply meta/main.yml:10]: Debian is added

File: handlers/main.yml (hunk 4 above)
Comment (tgadiev) [reply handlers/main.yml]: Where is it removed?

File: tasks/main.yml (hunk 27 above)
Comment (tgadiev) [reply tasks/main.yml]: Where?

**Review by tgadiev — CHANGES_REQUESTED** [review]

File: .travis.yml [hunk 32]
```diff
@@ -0,0 +1,26 @@
...
+  - "2.7"
+
+services:
+  - docker
+
+before_install:
+  - git clone https://github.com/lean-delivery/ansible-lint-rules.git ~/ansible-lint-rules
+
+install:
+  - pip install --upgrade ansible==2.5.* docker-py molecule==2.17 pyOpenSSL PyYAML pytest==3.7.4
```
Comment (tgadiev) [inline .travis.yml]:
```suggestion
  - pip install --upgrade ansible==2.7.* docker-py molecule==2.20 pyOpenSSL PyYAML pytest
```

File: tasks/main.yml [hunk 33]
```diff
@@ -0,0 +1,64 @@
...
+      - '{{ minio_home }}/bin'
+      - '{{ minio_data_path }}'
+    loop_control:
+      loop_var: dir
+
+  - name: Copy minio binary file
+    get_url:
+      url: '{{ minio_package_source }}'
+      dest: '{{ minio_home }}/bin/minio'
+      mode: +x
```
Comment (tgadiev) [inline tasks/main.yml:29]:
```suggestion
      mode: +x
    register: file_downloaded
    retries: 15
    delay: 5
    until: "'OK' in file_downloaded.msg"
```

File: tasks/main.yml [hunk 34]
```diff
@@ -0,0 +1,64 @@
...
+      state: directory
+      mode: 0750
+      owner: '{{ minio_user }}'
+      group: '{{ minio_group }}'
+    loop:
+      - '{{ minio_home }}'
+      - '{{ minio_home }}/bin'
+      - '{{ minio_data_path }}'
+    loop_control:
+      loop_var: dir
```
Comment (tgadiev) [inline tasks/main.yml]:
```suggestion
      loop_var: minio_dir
```

File: tasks/main.yml [hunk 35]
```diff
@@ -0,0 +1,64 @@
...
+- block:
+  - name: Create minio user
+    user:
+      name: '{{ minio_user }}'
+      home: '{{ minio_home }}'
+      shell: /sbin/nologin
+
+  - name: Ensure that minio directories are created
+    file:
+      path: '{{ dir }}'
```
Comment (tgadiev) [inline tasks/main.yml]:
```suggestion
      path: '{{ minio_dir }}'
```

**Review by tgadiev — CHANGES_REQUESTED** [review]

**Review by tgadiev — APPROVED** [review]

**Pull request merged**